hi-ml-cpath/src/health_cpath/configs/classification/BaseMIL.py

```diff
@@ -241,14 +253,33 @@ def get_transforms_dict(self, image_key: str) -> Optional[Dict[ModelKey, Union[C
     def get_encoder_params(self) -> EncoderParams:
         return create_from_matching_params(self, EncoderParams)
 
+    def get_num_classes(self) -> int:
+        """Gets the number of classes that the model handles. In training mode, this is the number of classes in the
+        training dataset. In evaluation mode, this is read directly from the `eval_num_classes` attribute.
+
+        This method has a side effect in training mode: It creates the data module if it is not yet set in the
+        `data_module` attribute.
+        """
+        if self.runner_mode == RunnerMode.EVAL_FULL:
+            # We are in evaluation mode, so we don't necessarilty have access to the training data module. The eval
+            # data module may not contain all classes, so we need to hardcode those values.
+            return self.eval_num_classes
```

Comment:
see comment above 

```suggestion
def get_num_classes(self) -> int:
        if self.num_classes is not None: 
             return self.num_classes 
        if (not hasattr(self, "data_module")) or self.data_module is None:
                if self.runner_mode == RunnerMode.EVAL_FULL:
                    self.data_module = self.get_eval_data_module()
                else:
                    self.data_module = self.get_data_module()
            return self.data_module.train_dataset.n_classes
```

Reply:
Thanks for that suggestion. I've first added it in, but then reverted. I wanted to keep the change limited to the problem we have now. For training, reading out all properties from the dataset is still the most flexible approach.